### PR TITLE
Fix application:env and stoping/starting related issues

### PR DESCRIPTION
**Pull request by yfyf**
This seems to work nicely.

However, I am a bit unhappy about the fact that there's a cyclic dependency between `piqi_rpc` and `piqi_rpc_http` modules. The fact that piqi-rpc listeners live in a different supervision tree opens up some more possibilities for race conditions, when adding/removing services and/or crashing, so it would be good to rethink this from scratch, but I'm too braindead for the moment to do it, so just pushing this out with a hope that someone else will spot any obvious issues if they're out there.

### CODE REVIEW

File: src/piqi_rpc_http.erl [hunk 1]
```diff
@@ -15,25 +15,27 @@
...
+-define(DEFAULT_NB_ACCEPTORS, 100).
 
 -include("piqi_rpc.hrl").
 
-start_link() ->
-    Routes = get_cowboy_routes(),
-    cowboy:start_http(get_http_env(name, ?DEFAULT_NAME), get_http_env(nb_acceptors, 100),
-        [{port, get_http_env(port, ?DEFAULT_PORT)}],
-        [{env, [{dispatch, Routes}]}]
+start_listeners() ->
```
Comment (stroop23) [inline src/piqi_rpc_http.erl]: start_listener(), there's only one

File: src/piqi_rpc.erl [hunk 2]
```diff
@@ -88,18 +108,27 @@ remove_service(RpcServiceDef) ->
     ok = remove_service_def(RpcService),
     ok = piqi_rpc_http:configure().
 
+-spec remove_all_services() -> ok.
+remove_all_services() ->
+    ok = piqi_rpc_http:cleanup(),
```
Comment (stroop23) [inline src/piqi_rpc.erl:113]: This will also stop the listener, i don't think we want to do that.

**Comment by yfyf** [comment]
Ugh, guess what, the weird race conditions I was seeing turned out to be a bug in ~~cowboy~~ ranch:

```
23> [begin {ok, _} = cowboy:start_http(foo, 100, [{port, 8080}] , []), ok = cowboy:stop_listener(foo), io:format("~p~n", [N]) end || N <- lists:seq(1, 10) ].
1
2
3
4
5
6
** exception error: no match of right hand side value {error,{shutdown,{child,undefined,
                                                                              {ranch_listener_sup,foo},
                                                                              {ranch_listener_sup,start_link,
                                                                                                  [foo,100,ranch_tcp,[{port,8080}],cowboy_protocol,[]]},
                                                                              permanent,5000,supervisor,
                                                                              [ranch_listener_sup]}}}
```

**Comment by yfyf** [comment]
ranch bug reported upstream: https://github.com/extend/ranch/issues/74

File: src/piqi_rpc.erl (hunk 2 above)
Comment (yfyf) [inline src/piqi_rpc.erl:113]: I changed `cleanup/0` to only unset the cowboy dispatching rules (and added a separate method for stopping the listener), so this is fixed now.

**Pull request merged**